pandas/io/parsers.py

```diff
@@ -2223,13 +2223,14 @@ def _clean_index_names(columns, index_col):
 def _get_empty_meta(columns, index_col, index_names):
     columns = list(columns)
 
-    if index_col is not None:
-        index = MultiIndex.from_arrays([[]] * len(index_col),
-                                       names=index_names)
-        for n in index_col:
-            columns.pop(n)
-    else:
+    if index_col is None or index_col is False:
         index = Index([])
+    else:
+        index_col = list(index_col)
+        index = MultiIndex.from_arrays([[]] * len(index_col), names=index_names)
+        index_col.sort()
```

Comment:
It needs sorting so that the calculated column index, i.e. `n-i` in the `pop`, evaluates to the correct value even for a multiindex that's specified with decrementing index columns. 

E.g. Without the sort and for `index_col=[1,0]`, the loop would try to `pop`  first column `1 - 0 = 0` and then `0 - 1 = -1`.

I expect I'm missing something but  `columns = columns - Index(index_col)` raises an error for me (`*** TypeError: can only perform ops with scalar values`)